docs/oss-k8s-e2e-setup.md

```diff
@@ -0,0 +1,941 @@
+--member="serviceAccount:$GSA@$PROJECT_ID.iam.gserviceaccount.com" \
+--role="roles/storage.admin"
+
+# Set WORKLOAD_IDENTITY_POOL to your own value.
+export WORKLOAD_IDENTITY_POOL=wi-pool-amacaskill-k8s-cluster
+gcloud iam workload-identity-pools create $WORKLOAD_IDENTITY_POOL \
+ --project=$PROJECT_ID \
+ --location="global" \
+ --display-name=$WORKLOAD_IDENTITY_POOL
+
+# This has to be run inside of the master vm (any kubectl commands do)
+gcloud compute ssh kub-m --zone=$ZONE --project=$PROJECT_ID
+
+sudo -s
+
+# Copy value of SUDO_USER to use below.
+echo "SUDO_USER is $SUDO_USER"
+
+exit
+sudo su -
+
+
+# This has to be run inside of the master vm (any kubectl commands do)
+kubectl get --raw /.well-known/openid-configuration | jq -r .issuer
+
+kubectl get --raw /openid/v1/jwks > cluster-jwks.json
+
+# move cluster-jwks.json from the master vm the dev machine in order to run the below gcloud commands.
+# IMPORTANT: replace value of SUDO_USER below value you copied above when sshed with sudo -s
+export S_USER=SUDO_USER
+sudo mv /root/cluster-jwks.json /home/$S_USER/
+
+sudo chown $S_USER:$S_USER /home/$S_USER/cluster-jwks.json
+
+exit
+exit
+
+gcloud compute scp kub-m:~/cluster-jwks.json . \
+    --zone=$ZONE \
+    --project=$PROJECT_ID   
+
+# Set WORKLOAD_IDENTITY_POOL_PROVIDER to your own value.
+export WORKLOAD_IDENTITY_POOL_PROVIDER=wi-p-amacaskill-k8s-cluster
+gcloud iam workload-identity-pools providers create-oidc $WORKLOAD_IDENTITY_POOL_PROVIDER \
+   --location="global" \
+   --workload-identity-pool="$WORKLOAD_IDENTITY_POOL" \
+   --issuer-uri="https://kubernetes.default.svc.cluster.local" \
+   --attribute-mapping="google.subject=assertion.sub" \
+   --display-name=$WORKLOAD_IDENTITY_POOL_PROVIDER \
+   --jwk-json-path="cluster-jwks.json"
+```
+
+## Create resources for Workload Identity
+
+Create a Kubernetes ServiceAccount for your application to use. You can also use any existing
+Kubernetes ServiceAccount in any namespace, including the default Kubernetes ServiceAccount.
+
+```bash
+# IMPORTANT: Replace namespace with the namespace you want to use.
+# Export outside VM for use in gcloud commands later.
+export NAMESPACE=gcsfuse-test-ns
+gcloud compute ssh kub-m --zone=$ZONE --project=$PROJECT_ID
+sudo su -
+
+# IMPORTANT: Replace namespace with the namespace you want to use.
+export NAMESPACE=gcsfuse-test-ns
+kubectl create namespace $NAMESPACE
+kubectl create serviceaccount k8s-on-gce-ksa --namespace $NAMESPACE
+
+# This is required because the driver uses this annotation to link the Kubernetes Service Account with a GCP SA in fetchGCPSAToken. This is so the driver can configure the KSA to impersonate IAM service accounts, which configures GKE to exchange the federated access token for an access token from the IAM Service Account Credentials API.
+kubectl annotate serviceaccount k8s-on-gce-ksa \
+    --namespace $NAMESPACE \
+iam.gke.io/gcp-service-account=$GSA@$PROJECT_ID.iam.gserviceaccount.com
```

Comment:
@amacaskill Add both skipCSIBucketAccess check and annotation method. Both are valid and customers might want to use for different ways.